### DISCUSSION TITLE
first time setting up this project.

### DISCUSSION THREAD

**Discussion by aniel300** [euzu/tuliprox#111] [Show and tell]
first time setting up this project. i think i got everything working (right now is simple) but i will come tomorrow to keep learning because am a little overwhelm. this is the last tine in logs. so am assuming everything is working `[2024-11-07T22:44:10Z INFO  m3u_filter::processing::playlist_processor] {"stats": {"name":"lion","type":"xtream","errors":0,"raw":{"groups":37,"channels":2989},"processed":{"groups":37,"channels":2989},"took":"0 secs"}}`

here are my current config in case i have something wrong

config.yml

```yml
#   sec  min   hour   day of month   month   day of week   year
schedule: "0  0  8,20  *  *  *  *"
web_ui_enabled: true
web_auth:
  enabled: false
  secret: very.secret.secret
  issuer: m3u_filter
  userfile: user.txt
  threads: 8
api: {host: 0.0.0.0, port: 8901, web_root: ./web}
working_dir: ./data
update_on_boot: true```

source.yml:

```yml
templates:
- name: ALL_CHAN
  value: 'Group ~ ".*"'
sources:
- inputs:
    - type: xtream
      headers:
        User-Agent: "Mozilla/5.0 (AppleTV; U; CPU OS 14_2 like Mac OS X; en-us) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.1 Safari/605.1.15"
        Accept: application/json
        Accept-Encoding: gzip
      url: 'http://xxx.com'
      username: xxx
      password: xxx
      name: lion
      # suffix:
      #   field: title
      #   value: (lion)
      options: {xtream_skip_vod: true, xtream_skip_series: true}

  targets:
    - name: test
      output:
        - type: xtream
          #filename: test.m3u
      sort:
        groups:
          order: asc
        channels:
          - {field: name, group_pattern: '^TR.:\s?(.*)', order: asc}
      filter: "!ALL_CHAN!"```

api-proxy.yml:

```yml
server:
  - name: default 
    protocol: http
    host: 0.0.0.0
    http_port: 80
    https_port: 443
    rtmp_port: 0
    timezone: Europe/Paris
    message: Welcome to m3u-filter
user:
  - target: test
    credentials:
      - {username: a, password: a, proxy: reverse}```

**Comment by euzu** [comment]
yes it worked. you can simply insert the credentials into an iptv app and test lt

**Reply by aniel300** [reply]
i will enable m3u output as well for the time being so it is easier to test on vlc

**Comment by euzu** [comment]
you can do both outputs. xc and m3u

**Comment by aniel300** [comment]
am getting this error `[2024-11-07T23:32:34Z ERROR actix_http::h1::dispatcher] stream error: request parse error: invalid Header provided`

**Reply by euzu** [reply]
doing what?

**Comment by aniel300** [comment]
trying to download/ reach the m3u

**Comment by euzu** [comment]
are you using some utf8 chars like chineese chars or something?

can you provide the url with anonymized credentials

**Reply by aniel300** [reply]
like my provider url or the output url?

**Comment by euzu** [comment]
i mean you tried to  to call an api from m3u-filter and got this error?
What did you call to get this error?
Or what did you do?

**Reply by aniel300** [reply]
i tried reaching it from my home internet to over to my remote server. i tried few configuration variations for the host and port including `    host: 0.0.0.0` and `    host: remote server public ip`. port i have tried changing from the default since port 80 and 443 are being used by my traefik

**Comment by euzu** [comment]
m3u-filter does not provide ssl termination.  
If you access m3u-filter from outside your network, you need to define the user for external access. 

![image](https://github.com/user-attachments/assets/83583007-f6d3-4dd2-bf8b-57a33fdf1cab)

And your external server configuration has to match your external access. Not the internal port.
Host: your dyndns  or external static ip
Port: your router port where your  nginx/caddy/traeffik  listens!   You need to forward the traffic to m3u-filter.

![image](https://github.com/user-attachments/assets/fce86ea6-9a3c-47a2-9a38-d0a1470688f7)

**Reply by aniel300** [reply]
ok it seems i was trying to use the default (which is for local access only) to access it remotely. adding another entry this time for external access fixed it. thank u for the support

**Comment by aniel300** [comment]
closing for now
